src/metabase/server/middleware/security.clj

```diff
@@ -124,11 +125,77 @@
           "Content-Security-Policy"
           #(format "%s frame-ancestors %s;" % (if allow-iframes? "*" (or (embedding-app-origin) "'none'")))))
 
+(defn parse-url
+  "Returns an object with protocol, domain and port for the given url"
+  [url]
+  (if (= url "*")
+    {:protocol nil :domain "*" :port "*"}
+    (let [pattern #"^(?:(https?)://)?([^:/]+)(?::(\d+|\*))?$"
+          matches (re-matches pattern url)]
+      (if matches
+        (let [[_ protocol domain port] matches]
+          {:protocol protocol
+           :domain domain
+           :port port})
+        (throw (IllegalArgumentException. (str "Invalid URL '" url "'")))))))
+
+
+(defn approved-domain?
+  "Checks if the domain is compatible with the reference one"
+  [domain reference-domain]
+  (or (= reference-domain "*")
```

Comment:
nit: indentation (arguments to `or` should be lined up)

Reply:
Thanks! Done in e8edb32d74015421e78825adc388ecc569aab50c